designs/instance-type-settings.md

```diff
@@ -0,0 +1,354 @@
+# Instance Type Settings
+
+## Goals
+
+Allow users to:
+
+* Define [custom device resources](https://github.com/aws/karpenter/pull/2161) on an instance type
```

Comment:
nit: Can we reference the issue instead of the pull request here? 
```suggestion
* Define [custom device resources](https://github.com/aws/karpenter/issues/1900) on an instance type
```